lib/matplotlib/backend_managers.py

```diff
@@ -245,6 +283,8 @@ def add_tool(self, name, tool, *args, **kwargs):
             else:
                 self._toggled.setdefault(tool_obj.radio_group, None)
 
+        if self.figure:
+            tool_obj.set_figure(self.figure)
```

Comment:
Yes the tools initialize with None, but when addding a tool, it is better to set the figure to the current figure of the Toolmanager, we could add a kwarg to `add_tool` to bypass the `tool.set_figure`  but I find it a little bit overkill